arrays/arrays.c

```diff
@@ -19,45 +19,56 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
+Array *create_array(int capacity)
+{
   // Allocate memory for the Array struct
-
+  Array *newarray = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  newarray->capacity = capacity;
+  newarray->count = 0;
   // Allocate memory for elements
+  newarray->elements = calloc(capacity, sizeof(char *));
 
+  return newarray;
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr)
+{
 
   // Free all elements
-
+  free(arr->elements);
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
+void *resize_array(Array *arr)
```

Comment:
How did an asterisk get in here? I strongly recommend against changing the return type of this expression. "void" expression means you want to just work with the existing input (Array *arr in this case) and just modify some attributes of it. In particular you will want to take the arr->elements and replace them with **newelements as you have begin to do down below. i.e. arr->elements = newelements.